src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/AccountsReferenceDateQuestionController.java

```diff
@@ -0,0 +1,146 @@
+
+        try {
+            companyProfile = companyService.getCompanyProfile(companyNumber);
+            smallFullAccounts = smallFullService.getSmallFullAccounts(apiClient, transactionId, companyAccountsId);
+        } catch (ServiceException e) {
+                LOGGER.errorRequest(request, e.getMessage(), e);
+                return ERROR_VIEW;
+        }
+
+        LocalDate companyProfilePeriodStartOn = companyProfile.getAccounts().getNextAccounts().getPeriodStartOn();
+        LocalDate companyProfilePeriodEndOn = companyProfile.getAccounts().getNextAccounts().getPeriodEndOn();
+
+        LocalDate smallFullPeriodEndOn = smallFullAccounts.getNextAccounts().getPeriodEndOn();
+
+        if (companyProfilePeriodEndOn != smallFullPeriodEndOn) {
+            //set radio button choice to NO - As the user has changed them so the date must not be the correct date.
+        } else {
+            setHasConfirmedAccountingReferenceDate(request, accountsReferenceDateQuestion);
+        }
+
+        model.addAttribute(ACCOUNTS_REFERENCE_DATE_QUESTION, accountsReferenceDateQuestion);
+        model.addAttribute("startDate", companyProfilePeriodStartOn);
+        model.addAttribute("endDate", companyProfilePeriodEndOn);
+
+        return getTemplateName();
+    }
+
+    @PostMapping
+    public String submitAccountsReferenceDateQuestion(@PathVariable String companyNumber,
+                                                @PathVariable String transactionId,
+                                                @PathVariable String companyAccountsId,
+                                                @ModelAttribute(ACCOUNTS_REFERENCE_DATE_QUESTION) @Valid AccountsReferenceDateQuestion accountsReferenceDateQuestion,
+                                                BindingResult bindingResult,
+                                                HttpServletRequest request) {
+
+        if (bindingResult.hasErrors()) {
+            return getTemplateName();
+        }
+
+        CompanyProfileApi companyProfile;
+        SmallFullApi smallFullAccounts;
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        try {
+            companyProfile = companyService.getCompanyProfile(companyNumber);
+            smallFullAccounts = smallFullService.getSmallFullAccounts(apiClient, transactionId, companyAccountsId);
+        } catch (ServiceException e) {
+            LOGGER.errorRequest(request, e.getMessage(), e);
+            return ERROR_VIEW;
+        }
+
+        LocalDate companyProfilePeriodEndOn = companyProfile.getAccounts().getNextAccounts().getPeriodEndOn();
+        LocalDate smallFullPeriodEndOn = smallFullAccounts.getNextAccounts().getPeriodEndOn();
+
+        if (accountsReferenceDateQuestion.getHasConfirmedAccountingReferenceDate()) {
+            if (companyProfilePeriodEndOn != smallFullPeriodEndOn) {
+                try {
+                    smallFullService.updateSmallFullAccounts(smallFullAccounts, transactionId, companyAccountsId );
```

Comment:
Logic to get to this point is bang on. The only thing I would say is at the mo we may make up to 3 API calls:
- GET company profile
- GET small full
- PUT small full

Question - and by no means a recommendation, just opening the conversation - is there any harm in just performing a PUT on the small full resource if they've selected the yes radio button? The date may already be correct on the small full resource, so the PUT wouldn't do anything per se, but we'd cut down from 3 API calls to just 1 in that scenario

Reply:
Great recommendation, will look to implement this.